templates/types/streaming/nextjs/app/components/ui/chat/chat-input.tsx

```diff
@@ -19,30 +22,70 @@ export default function ChatInput(
     | "handleSubmit"
     | "handleInputChange"
     | "messages"
+    | "setInput"
+    | "append"
   >,
 ) {
   const [imageUrl, setImageUrl] = useState<string | null>(null);
   const { files, uploadNew, removeFile, resetUploadedFiles } = useCsv(
     props.messages,
   );
 
-  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+  const getAttachments = () => {
+    if (!imageUrl && files.length === 0) return undefined;
+    const annotations: MessageAnnotation[] = [];
     if (imageUrl) {
-      props.handleSubmit(e, {
-        data: { imageUrl: imageUrl },
+      annotations.push({
```

Comment:
AFAIK the server can also send user annotations. Now we're mixing creating annotations on the server side and on the client side, which might be confusing